src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -57,28 +66,18 @@ public LineResponse updateById(Long id, LineRequest lineRequest) {
 
     @Transactional
     public void deleteById(Long id) {
-        checkExistLine(lineRepository.findById(id), id);
+        getLine(id);
         lineRepository.deleteById(id);
```

Comment:
```suggestion
        lineRepository.delete(getLine(id));
```
deleteById는 내부적으로 id를 가지고 엔티티를 조회하기 때문에
line을 조회했다면 이렇게 작성해도 될거같네요! 😀
```java
@Transactional
@Override
public void deleteById(ID id) {

Assert.notNull(id, ID_MUST_NOT_BE_NULL);

delete(findById(id).orElseThrow(() -> new EmptyResultDataAccessException(
		String.format("No %s entity with id %s exists!", entityInformation.getJavaType(), id), 1)));
}
```

Reply:
오호 이건 몰랐던 사실이네요! 내부적으로 select -> delete 구조이니 앞서 select를 했다면 그냥 delete만 할 수 있도록!